sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/complexTypesSuite.scala

```diff
@@ -439,4 +439,17 @@ class ComplexTypesSuite extends PlanTest with ExpressionEvalHelper {
         .select('c as 'sCol2, 'a as 'sCol1)
     checkRule(originalQuery, correctAnswer)
   }
+
+  test("SPARK-24313: support complex types as map keys") {
+    val mb0 = Literal.create(
+      Map(Array[Byte](1, 2) -> "1", Array[Byte](3, 4) -> null, Array[Byte](2, 1) -> "2"),
+      MapType(BinaryType, StringType))

```

Comment:
shall we test `ArrayType`to reflect the test name?

Reply:
I changed the test name here and the comment for `ElementAt` because I found an issue in using arrays as keys for map. When I do that, I get:
```
Caused by: java.lang.ClassCastException: scala.collection.immutable.$colon$colon cannot be cast to org.apache.spark.sql.catalyst.util.ArrayData
```
I will investigate this further, but I think this is a separate issue and should be addressed in another PR, what do you think?